packages/extension-browser/src/content-script/connector.ts

```diff
@@ -69,31 +70,14 @@ export default class WebExtensionConnector implements IConnector {
         browser.runtime.sendMessage(message);
     }
 
-    /** Traverses the DOM while sending `element::*` events. */
-    private async traverseAndNotify(node: Element, doc: IAsyncHTMLDocument): Promise<void> {
-        const element = new AsyncHTMLElement(node, doc);
-        const name = node.tagName.toLowerCase();
-        const resource = location.href;
-
-        await this._engine.emitAsync(`element::${name}` as 'element::*', { element, resource });
-        await this._engine.emitAsync(`traverse::down`, { element, resource });
-
-        // Recursively traverse child elements.
-        for (let i = 0; i < node.children.length; i++) {
-            await this.traverseAndNotify(node.children[i], doc);
-        }
-
-        await this._engine.emitAsync(`traverse::up`, { element, resource });
-    }
-
     private setFetchElement(event: FetchEnd) {
         const url = event.request.url;
-        const elements = Array.from(document.querySelectorAll('[href],[src]')).filter((element: any) => {
+        const elements = Array.from(this._document.querySelectorAll('[href],[src]')).filter((element: any) => {
```

Comment:
You also needed to resolve the original element for `FetchEnd` in other connectors too, right? Should this be a shared helper? If so, we may also need to include `poster`, `data`, and `srcset` for completeness.

Reply:
Are you thinking in a new `util` in `hint`?

Reply:
I'm taking a look to this and they are not compatible (at least not now). In jsdom, we have the element that is fetching the content, but not in the browser extension.

Also, I found another problem here, in case of a relative url, event.request.url will contain the full url (domain + relative url), but the element will contain the relative url, so this code won't work.

I'm thinking in something like `url.endsWith(element.getAttribute('src')) || ....`.

Maybe there is another way, but I don't know this code too much.

What do you think @antross?

Reply:
> If so, we may also need to include `poster`, `data`, and `srcset` for completeness.

@antross what elements use the attribute 'data'

Reply:
[From `https-only`](https://github.com/webhintio/hint/blob/3dc05691a36fd8d46f32634d964796c68440432c/packages/hint-https-only/src/hint.ts#L124):
```js
/*
 * Possible URL sources:
 *
 * * <img>, <audio>, <video>, <source> and <track> can have a `src` attribute.
 *   E.g.: <img src="http://example.com/image.jpg">
 *
 * * <img> and <source> can have a `srcset` attribute.
 *   https://developer.mozilla.org/en-US/docs/Web/HTML/Element/source
 *   E.g.: <img src="image-src.png" srcset="image-1x.png 1x, image-2x.png 2x">
 *
 * * <video>  can have a `poster` attribute.
 *   E.g.: <video width="480" controls poster="https://archive.org/download/WebmVp8Vorbis/webmvp8.gif" />
 *
 * * <object> has a `data` attribute.
 *   E.g.: <object data="movie.swf" type="application/x-shockwave-flash"></object>
 *   https://developer.mozilla.org/en-US/docs/Web/HTML/Element/object
 */
```

There's also `srcset` that uses another format that we should probably take into consideration.

Reply:
I'm taking into consideration this format `image-1x.png 1x, image-2x.png 2x`. There is any other format?